### ISSUE TITLE
[Research] Real-Time Empathy and Social Cognition Detection from Frontal EEG

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#126]
## Research Topic
Real-time empathy and theory of mind (ToM) detection from frontal EEG using mu rhythm suppression, mirror neuron system activation, and inter-brain synchronization (hyperscanning).

## Paper References

1. **"Exploring Empathic Engagement in Immersive Media: An EEG Study on Mu Rhythm Suppression in VR"** — PLOS ONE, May 2024. Key finding: **Mu rhythm (8-13 Hz) suppression at frontal and central sites** correlates with empathic engagement. Individuals with stronger mu suppression reported higher empathy scores. VR context enhanced mu suppression compared to traditional media. Demonstrates that mu-ERD is a viable real-time empathy biomarker.

2. **"From Neural Networks to Emotional Networks: A Systematic Review of EEG-Based Emotion Recognition in Cognitive Neuroscience and Real-World Applications"** — Sensors, March 2025. Comprehensive review linking emotion recognition, empathy, and social cognition in EEG. Empathic responses activate similar EEG patterns as first-person emotional experiences but with reduced amplitude + additional self-other distinction activity. People with high empathy show **significantly lower P300 amplitudes** — better emotional processing efficiency.

3. **"Neural Empathy Mechanisms Are Shared for Physical and Social Pain, and Increase from Adolescence to Older Adulthood"** — Social Cognitive and Affective Neuroscience, 2024. Shared neural empathy mechanisms for physical and social pain with age-dependent maturation. EEG signatures of empathy (frontal theta, mu suppression) strengthen with age, suggesting trainability.

4. **"Mapping EEG Metrics to Human Affective and Cognitive Models: An Interdisciplinary Scoping Review"** — Biomimetics, 2025. Maps EEG metrics to cognitive/affective models including empathy and theory of mind. Frontal midline theta (4-8 Hz) increases with cognitive empathy (perspective-taking). Alpha suppression at temporal sites correlates with emotional empathy (feeling with others).

5. **"Current Trends in Hyperscanning Research"** — mBrainTrain, 2024-2025. EEG hyperscanning (dual-brain recording) enables real-time monitoring of inter-brain synchronization during social interaction. Decreased inter-brain synchronization in right middle frontal cortex during certain social conditions. Portable/consumer EEG makes hyperscanning feasible outside the lab.

## Application to NeuralDreamWorkshop

This introduces a social cognition module — detecting empathic engagement in real-time. Use cases:
- **Couples therapy**: monitor empathic resonance between partners (hyperscanning with 2 Muse devices)
- **Media empathy**: measure empathic engagement while watching emotional content
- **Social skills training**: real-time feedback on empathic processing
- **Meditation**: loving-kindness meditation depth via empathy-related EEG markers

The system already has frontal alpha/theta features (AF7/AF8) and temporal channels (TP9/TP10). Mu rhythm is detectable at these sites, though not as cleanly as at C3/C4.

### New files:
- `ml/models/empathy_detector.py` — EmpathyDetector class
- `ml/processing/mu_rhythm.py` — mu rhythm extraction and ERD computation
- `ml/models/hyperscanning.py` — InterBrainSync class (dual Muse)
- `client/src/pages/empathy-monitor.tsx` — empathy detection dashboard

### Key classes/methods:
```python
class EmpathyDetector:
    """Detects empathic engagement from 4-channel Muse 2 EEG."""
    
    def __init__(self):
        self.baseline_mu = None  # resting mu power (calibrated)
        self.ema_alpha = 0.3     # smoothing for real-time output
        
    def extract_empathy_features(self, eeg_4ch: np.ndarray, fs: int = 256) -> dict:
        """Returns:
        - mu_suppression_af7: float (mu ERD at left frontal)
        - mu_suppression_af8: float (mu ERD at right frontal)
        - mu_suppression_index: float (averaged, baseline-normalized)
        - frontal_theta_increase: float (cognitive empathy marker)
        - temporal_alpha_suppression: float (emotional empathy marker)
        - theta_gamma_coupling: float (empathic depth)
        - self_other_distinction: float (beta rebound after mu suppression)
        """
    
    def predict_empathy_state(self, features: dict) -> dict:
        """Returns: empathy_score (0-1), empathy_type (cognitive/emotional/both),
                    engagement_level (low/medium/high), 
                    mirror_neuron_activation_estimate"""
    
    def calibrate_baseline(self, resting_eeg: np.ndarray, fs: int = 256):
        """Record 2-min resting mu power for ERD computation."""

class MuRhythmExtractor:
    """Extract mu rhythm (8-13 Hz) specifically for mirror neuron system monitoring."""
    
    def compute_mu_power(self, signal: np.ndarray, fs: int = 256) -> float:
        """Bandpass 8-13 Hz, compute Welch PSD in mu band."""
    
    def compute_mu_erd(self, signal: np.ndarray, baseline: np.ndarray, fs: int) -> float:
        """Event-related desynchronization: (baseline - active) / baseline * 100."""
    
    def separate_mu_from_alpha(self, eeg_4ch: np.ndarray, fs: int) -> dict:
        """Attempt to separate mu (sensorimotor) from occipital alpha.
        With Muse 2 (no C3/C4/O1/O2), this is approximate at best.
        Uses reactivity to motor imagery vs eyes-open/closed."""

class InterBrainSync:
    """Hyperscanning: inter-brain synchronization between two Muse 2 devices."""
    
    def compute_ibs(self, eeg_person1: np.ndarray, eeg_person2: np.ndarray,
                     fs: int = 256, band: str = 'alpha') -> dict:
        """Returns: plv_matrix (4x4 cross-brain), mean_sync, 
                    frontal_sync (AF7-AF7, AF8-AF8), temporal_sync"""
    
    def detect_empathic_resonance(self, sync_timeseries: list[float]) -> dict:
        """Identifies moments of high inter-brain synchronization 
        (putative empathic resonance episodes)."""
```

### Empathy metrics for UI:
- **Empathy Score** (0-100): composite of mu suppression + theta increase + temporal alpha
- **Empathy Type**: cognitive (perspective-taking, theta-dominant) vs emotional (feeling-with, mu-dominant)
- **Mirror Neuron Activation**: mu-ERD strength (percentage below baseline)
- **Engagement Timeline**: continuous empathy score over time during content viewing
- **Hyperscanning View** (dual Muse): inter-brain synchronization heatmap

## Expected Accuracy/Impact
- Mu suppression detection: **feasible at AF7/AF8** but 30-40% weaker than C3/C4
- Empathy high/low binary classification: **60-70%** (limited by electrode placement)
- Hyperscanning inter-brain sync: detectable but very noisy with consumer EEG
- The 2024 VR paper shows mu suppression correlates with self-reported empathy (r=0.3-0.5)
- Main limitation: mu rhythm at frontal sites overlaps heavily with posterior alpha — separation is approximate
- Value: unique feature, no consumer EEG app currently offers empathy monitoring

## Complexity Rating: HIGH
- Mu rhythm extraction at AF7/AF8 is noisy — not over motor cortex
- Separating mu from posterior alpha requires clever signal processing (reactivity-based)
- Hyperscanning requires BrainFlow dual-device support + time synchronization
- P300 detection requires event-related protocol (stimulus timing) — adds UI complexity
- Empathy ground truth labels are hard to obtain — self-report is unreliable
- EMG contamination at frontal sites affects mu band (8-13 Hz overlaps with EMG)
- Novel feature with limited validation on consumer EEG — should be labeled "experimental"
